test/jest/test.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const { Client } = require('../..')
+const { createServer } = require('http')
+/* global test, expect */
+
+test('should work in jest', async () => {
+  const server = createServer((req, res) => {
+    expect(req.url).toBe('/')
+    expect(req.method).toBe('POST')
+    expect(req.headers.host).toBe('localhost')
+    res.setHeader('Content-Type', 'text/plain')
+    res.end('hello')
+  })
+  await new Promise((resolve, reject) => {
+    server.listen(0, async () => {
```

Comment:
please just wrap server listen in a Promise, and do not use async functions inside new Promise.